pvlib/modelchain.py

```diff
@@ -924,6 +933,41 @@ def fuentes_temp(self):
             self.weather['wind_speed'])
         return self
 
+    @property
+    def dc_ohmic_model(self):
+        return self._dc_ohmic_model
+
+    @dc_ohmic_model.setter
+    def dc_ohmic_model(self, model):
+        if model is None:
+            self._dc_ohmic_model = self.no_dc_ohmic_loss
+        elif isinstance(model, str):
+            model = model.lower()
+            if model == 'dc_ohms_from_percent':
+                self._dc_ohmic_model = self.dc_ohms_from_percent
+            elif model == 'no_loss':
+                self._dc_ohmic_model = self.no_dc_ohmic_loss
+            else:
+                raise ValueError(model + ' is not a valid losses model')
+        else:
+            self._dc_ohmic_model = partial(model, self)
+
+    def dc_ohms_from_percent(self):
+        """
+        Calculate time series of ohmic losses and apply those to the mpp power
+        output of the `dc_model` based on the pvsyst equivalent resistance
+        method. Uses a `dc_ohmic_percent` parameter in the `losses_parameters`
+        of the PVsystem.
+        """
+        Rw = self.system.dc_ohms_from_percent()
```

Comment:
I don't think we want to embed a model in a ModelChain method. I would put this calculation in a new function `pvsystem.dc_ohmic_loss`, and have the `ModelChain` method call that function.

Reply:
I thought about that and @wholmgren had the same comment, but there are other calculations that run as `ModelChain` methods, such as 'pvwatts_losses' (though the calculations there are slightly simpler). I modeled this off of that, where it calls a `'pvsystem` method to retrieve a parameter based on system specifications, but then operates on `ModelChain` attributes. This method only makes sense in the context of a `ModelChain` where you have dc data produced by the chain. It felt awkward to pass a `ModelChain` attribute (`dc`) into a `pvsystem` function, and have that function only be used by `ModelChain`. But if that is the preference I am happy to make the change.

Reply:
Definitely right to say it would be awkward to pass a `ModelChain` attribute into a `pvsystem` function or `PVSystem` method, but I don't think that's required. A couple of ideas...

First, I think we should add this trivial function:

```python
# in pvsystem.py
def dc_ohmic_losses(ohms, current):
    return ohms * current * current
```

Then your `ModelChain` method could be:

```python
    def dc_ohms_from_percent(self):
        Rw = self.system.dc_ohms_from_percent()
        self.dc_ohmic_losses = pvsystem.dc_ohmic_losses(Rw, self.dc['i_mp'])
        self.dc['p_mp'] = self.dc['p_mp'] - self.dc_ohmic_losses
        return self
```

Or add a new `PVSystem` method:

```python
# pvsystem.py
    def dc_ohmic_losses_from_percent(self, current):
        Rw = self.dc_ohms_from_percent()
        return pvsystem.dc_ohmic_losses(Rw, current)

# modelchain.py
    def dc_ohms_from_percent(self):
        self.dc_ohmic_losses = self.system.dc_ohmic_losses(self.dc['i_mp'])
        self.dc['p_mp'] = self.dc['p_mp'] - self.dc_ohmic_losses
        return self
```